### PR TITLE
Chore: Refactor CustomEmoji to Typescript

### PR DESCRIPTION
**Pull request by liversonp**
Co-authored-by: guilhermedfs <guilhermedfs11@gmail.com>

## Proposed changes

Change emoji file from Javascript to Typescript

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] Improvement (non-breaking change which improves a current function)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Documentation update (if none of the other choices apply)

## Checklist

- [ ] I have read the [CONTRIBUTING](https://github.com/RocketChat/Rocket.Chat/blob/develop/.github/CONTRIBUTING.md#contributing-to-rocketchat) doc
- [ ] I have signed the [CLA](https://cla-assistant.io/RocketChat/Rocket.Chat.ReactNative)
- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works (if applicable)
- [ ] I have added necessary documentation (if applicable)
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by diegolmello — COMMENTED** [review]
Hey, thanks for your contribution!
What's the benefit of migrating models to TS?
I don't think we would actually use these types on schemes or on queries.

What we use really is the `CUSTOM_EMOJIS_TABLE`, which is going to be part of operations and it was introduced here https://github.com/RocketChat/Rocket.Chat.ReactNative/pull/3580

For the sake of migrating all files to Typescript, I'd say it's a good thing to migrate all models, but then it would require a different lint.
Otherwise it would be too cumbersome to type record on several files.

In other words, we could migrate all models, but without having to type `@field('name') name;` to `string | undefined`.
I'm ok with a simple ts-ignore for the whole file.
